cmd/resource-topology-exporter/main_test.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/k8stopologyawareschedwg/resource-topology-exporter/pkg/podrescli"
+)
+
+var (
+	pArgs *progArgs
+	err   error
+)
+
+var _ = Describe("Test argument parsing", func() {
+	BeforeEach(func() {
+		pArgs = nil
+	})
+	Context("when parsing command line arguments", func() {
+		It("contains correct values in flags provided", func() {
+			pArgs, err = parseArgs("--oneshot", "--no-publish")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pArgs.nrtupdater.Oneshot).To(BeTrue())
+			Expect(pArgs.nrtupdater.NoPublish).To(BeTrue())
+		})
+
+		It("must parse custom flags properly", func() {
+			pArgs, err = parseArgs("--kubelet-state-dir=dir1 dir2 dir3", "--reference-container=ns/pod/cont")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pArgs.resourcemonitor.KubeletStateDirs).To(BeEquivalentTo([]string{"dir1", "dir2", "dir3"}))
+			Expect(pArgs.rte.ReferenceContainer).To(Equal(&podrescli.ContainerIdent{Namespace: "ns", PodName: "pod", ContainerName: "cont"}))
+		})
+
+		It("should have the following default values", func() {
```

Comment:
mixed feelings here. We should probably keep this test, but only after having added a way to initialize all the structs 
with default values (without, somehow, duplicating the values we set from flags), and use something the golden file pattern (see: https://about.sourcegraph.com/go/advanced-testing-in-go/#Golden-files-also-test-flags)

IOW: the default values should be taken from a file, not buried in code.

Reply:
Cool I'll try that